server/src/test/java/io/spine/examples/chatspn/server/e2e/TestUser.java

```diff
@@ -0,0 +1,200 @@
+
+    public TestUser() {
+        channel = forAddress(ADDRESS, port())
+                .usePlaintext()
+                .build();
+        client = usingChannel(channel).build();
+        user = registerUser();
+    }
+
+    public UserId userId() {
+        return user.getId();
+    }
+
+    public EmailAddress email() {
+        return user.getEmail();
+    }
+
+    public UserProfile profile() {
+        return user;
+    }
+
+    public ChatPreview createPersonalChatWith(UserId member) {
+        CreatePersonalChat command = createPersonalChat(userId(), member);
+        postCommand(command);
+        ChatPreview chatPreview = chatPreview(command);
+        return chatPreview;
+    }
+
+    public MessageView sendMessageTo(ChatId chat) {
+        SendMessage command = sendMessage(chat, userId());
+        postCommand(command);
+        MessageView messageView = messageView(command);
+        return messageView;
+    }
+
+    public MessageView editMessage(MessageView message) {
+        EditMessage command = editMessageCommand(message);
+        postCommand(command);
+        MessageView messageView = messageView(command);
+        return messageView;
+    }
+
+    public void removeMessage(MessageView message) {
+        RemoveMessage command = removeMessageCommand(message);
+        postCommand(command);
+    }
+
+    public void deleteChat(ChatId chat) {
+        DeleteChat command = deleteChatCommand(chat, userId());
+        postCommand(command);
+    }
+
+    public UserChats readChats() {
+        ImmutableList<UserChats> userChatsList = client
+                .onBehalfOf(userId())
+                .select(UserChats.class)
+                .byId(userId())
+                .run();
+        return userChatsList.get(0);
+    }
+
+    public ImmutableList<MessageView> readMessagesIn(ChatId chat) {
+        String chatField = MessageView.Field
+                .chat()
+                .getField()
+                .toString();
+        String whenPostedField = MessageView.Field
+                .whenPosted()
+                .getField()
+                .toString();
+        ImmutableList<MessageView> userChatsList = client
+                .onBehalfOf(userId())
+                .select(MessageView.class)
+                .where(eq(new EntityColumn(chatField), chat))
+                .orderBy(new EntityColumn(whenPostedField), ASCENDING)
+                .run();
+        return userChatsList;
+    }
+
+    public UserProfile findUserBy(EmailAddress email) {
+        String emailField = UserProfile.Field
+                .email()
+                .getField()
+                .toString();
+        ImmutableList<UserProfile> profiles = client
+                .onBehalfOf(userId())
+                .select(UserProfile.class)
+                .where(eq(new EntityColumn(emailField), email))
+                .run();
+        return profiles.get(0);
+    }
+
+    public void closeConnection() throws InterruptedException {
```

Comment:
Isn't this something like `leave()`? It does not look a proper endpoint for `TestUser`.

I also don't understand the idea of exposing the exception. What are we expecting users of this API doing with it?